File: boilerplate/java/order-service/src/main/java/com/example/orderservice/infrastructure/security/BCryptPasswordHasher.java
package com.example.orderservice.infrastructure.security;

import com.example.orderservice.domain.ports.PasswordHasher;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class BCryptPasswordHasher implements PasswordHasher {

    private final BCryptPasswordEncoder encoder;

    public BCryptPasswordHasher() {
        this.encoder = new BCryptPasswordEncoder();
    }

    @Override
    public String hash(String plaintext) {
        return encoder.encode(plaintext);
    }

    @Override
    public boolean matches(String plaintext, String hashed) {
        return encoder.matches(plaintext, hashed);
    }
}

File: boilerplate/python/order-service/src/infrastructure/security/bcrypt_password_hasher.py
from __future__ import annotations
from domain.ports.auth_ports import PasswordHasher
import bcrypt

class BCryptPasswordHasher(PasswordHasher):
    def hash(self, plaintext: str) -> str:
        salt = bcrypt.gensalt()
        hashed = bcrypt.hashpw(plaintext.encode('utf-8'), salt)
        return hashed.decode('utf-8')

    def matches(self, plaintext: str, hashed: str) -> bool:
        try:
            return bcrypt.checkpw(plaintext.encode('utf-8'), hashed.encode('utf-8'))
        except Exception:
            return False
